docs/content/Dialog2.mdx

```diff
@@ -105,8 +105,80 @@ The `DialogHeaderProps` interface extends `DialogProps` and adds these additiona
+| autoFocus  | `boolean`                               | false    | If true, this button will be automatically focused when the dialog is first rendered. If `autoFocus` is true on subsequent button definitions, it will be ignored. |
+
+## ConfirmationDialog
+
+A `ConfirmationDialog` is a special kind of dialog with more rigid behavior. It is used to confirm a user action. `ConfirmationDialog`s always have exactly two buttons: one to cancel the action and one to confirm it. No custom rendering capabilities are provided for ConfirmationDialog.
+
+### useConfirm hook
+
+An alternate way to use `ConfirmationDialog` is with the `useConfirm()` hook. It takes no parameters and returns an `async` function, `confirm`. When `confirm` is called (see ConfirmOptions below for its argument), it shows the confirmation dialog. When the dialog is dismissed, a promise is resolved with `true` or `false` depending on whether or not the confirm button was used.
+
+### Example (with `useConfirm` hook)
+
+```javascript live noinline
+function ShorthandExample2() {
+  const confirm = useConfirm()
+  const buttonClick = React.useCallback(
+    async function (e) {
+      // @todo The docs site live code examples don't work with async/await, so using the raw promise instead!
+      if (await confirm({title: 'Are you sure?', description: 'You must confirm this action to continue'})) {
+        e.target.textContent = 'Confirmed!'
+      }
+    },
+    [confirm]
+  )
+  return (
+    <>
+      <Button onClick={buttonClick}>Confirmable action</Button>
+    </>
+  )
+}
+render(<ShorthandExample2 />)
+```
+
+### Example (using the full `ConfirmationDialog` component)
+
+```jsx live
+<State default={false}>
+  {([isOpen, setIsOpen]) => {
+    const openDialog = React.useCallback(() => setIsOpen(true), [setIsOpen])
+    const closeDialog = React.useCallback(() => setIsOpen(false), [setIsOpen])
+    return (
+      <>
+        <Button onClick={openDialog}>Open</Button>
+        {isOpen && (
+          <ConfirmationDialog title="Confirm action?" onClose={closeDialog}>
+            Are you sure you want to confirm this action?
+          </ConfirmationDialog>
+        )}
+      </>
+    )
+  }}
+</State>
+```
+
+### ConfirmationDialogProps
+
+| Prop name          | Type                                                                  | Default    | Description                                                                                                                   |
+| :----------------- | :-------------------------------------------------------------------- | :--------- | :---------------------------------------------------------------------------------------------------------------------------- |
+| title              | `string`                                                              |            | Required. Sets the title of the dialog, which by default is also used as the `aria-labelledby` attribute.                     |
```

Comment:
has the behavior of assigning the title to `aria-labelledby` been implemented? Can't see where that is happening in either Dialog or ConfirmationDialog

Reply:
It happens here: https://github.com/primer/components/blob/main/src/Dialog/Dialog.tsx#L286

Were you expecting something else?

Reply:
whoops, must have missed that!